pkg/expressions/cel.go

```diff
@@ -0,0 +1,135 @@
+package expressions
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/cel-go/cel"
+
+	harness "github.com/kudobuilder/kuttl/pkg/apis/testharness/v1beta1"
+)
+
+func buildProgram(expr string, env *cel.Env) (cel.Program, error) {
+	ast, issues := env.Compile(expr)
+	if issues != nil && issues.Err() != nil {
+		return nil, fmt.Errorf("type-check error: %s", issues.Err())
+	}
+
+	prg, err := env.Program(ast)
+	if err != nil {
+		return nil, fmt.Errorf("program construction error: %w", err)
+	}
+
+	return prg, nil
+}
+
+func buildEnv(resourceRefs []harness.TestResourceRef) (*cel.Env, error) {
+	env, err := cel.NewEnv()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create environment: %w", err)
+	}
+
+	for _, resourceRef := range resourceRefs {
+		env, err = env.Extend(cel.Variable(resourceRef.Ref, cel.DynType))
+		if err != nil {
+			return nil, fmt.Errorf("failed to add resource parameter '%v' to environment: %w", resourceRef.Ref, err)
+		}
+	}
+
+	return env, nil
+}
+
+// RunAssertExpressions evaluates a set of CEL expressions specified as AnyAllExpressions
+func RunAssertExpressions(
```

Comment:
There is little common code between the `all` and `any` case, let's split this into two separate functions.